zeth-contracts/contracts/BaseMixer_tests.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity ^0.5.0;
+
+import './BaseMixer.sol';
+
+contract BaseMixer_tests {
```

Comment:
Throughout this PR, and https://github.com/clearmatics/zeth/pull/104 we bumped into several edge cases that should be fixed now:
1. The `end` pointer (now removed) was potentially smaller than `start`
2.  The `require` on `length` was not properly set

Thus, for the tests to be meaningful, we need to think about these few edge cases and trigger them in the tests to make sure that the flaws patched in the PR have properly been patched.
Hence, please add test cases to hit the `require`, to pass a length == 0 (hit the return), and other potential scenarios you feel are relevant to test the fixes discussed in this PR

Reply:
I totally agree.
I intend to add a few more tests. 

I want to cover the following cases:
- length of 0
- length within the `require`
- length above the `require`
And for length within the `require`:
- bits spread over two field elements
- bits spread over one field element but two bytes
- bits spread over one byte of a field element.

I also intend to add a `require` on the size of  the function input `primary_inputs` (we might try to retrieve non existing elements) and add some tests on it.

Reply:
I forgot: we also need to constraint `start` and `start + length - 1`.